Dockerfile

```diff
@@ -25,23 +25,23 @@ COPY . .
 
 # Build the application, with optimizations, with static linking, and using jemalloc
 # N.B.: The static version of jemalloc is incompatible with the static Swift runtime.
-RUN swift build -c release \
+RUN mkdir /staging
+RUN --mount=type=cache,target=/build/.build \
+    swift build -c release \
         --product {{name}} \
         --static-swift-stdlib \
-        -Xlinker -ljemalloc
+        -Xlinker -ljemalloc && \
+    # Copy main executable to staging area
+    cp "$(swift build --package-path /build -c release --show-bin-path)/{{name}}" /staging && \
+    # Copy static swift backtracer binary to staging area
+    cp "/usr/libexec/swift/linux/swift-backtrace-static" /staging && \
+    # Copy resources bundled by SPM to staging area
+    find -L "$(swift build --package-path /build -c release --show-bin-path)/" -regex '.*\.resources$' -exec cp -Ra {} /staging \;
```

Comment:
Can we at least put some of the blank lines back in to space these out like they used to be, even if we can't have them neatly separated into different commands anymore? (Which I find terribly annoying, but I'm assuming it's related to making the caching work.)